app/views/posts/show.html.erb

```diff
@@ -8,13 +8,21 @@
       <p class="card-text"><%= @post.text %></p>
       <div class="col-md-3 text-md-end col-sm-3" style="width: 100%;">
         <p class="text-muted mt-5"><span>Likes: <%= @post.like_counter %></span> <span>Comments: <%= @post.comment_counter %></span></p>
+        <% if (can? :destroy, @post) %>
+        <%= button_to "DELETE Post", user_post_path(@user, @post), method: :delete, class: "btn btn-danger" %>
+
+        <% end %>
       </div>
     </div>
   </div>
   <div class="card mb-5" style="width: 100%;">
   <% @post.comments.each do |comment| %>
     <div class="card-body">
       <p class="card-text"><%= comment.user.name %>: <%= comment.text %></p>
+      <% if (can? :destroy, comment) %>
+      <%= button_to "DELETE Comment", user_post_path(@user, @post), method: :delete, class: "btn btn-danger" %>
+
+      <% end %>
```

Comment:
- [ ] The **_URL helper_** you are using for deleting the comment is wrong and this means that you won't be able to delete a comment but the post itself when the `DELETE Comment` button is pressed. This will cause unexpected UI interactions in the application. Kindly use the correct **_URL helper_**. 👍 😎